Firestore/core/src/remote/serializer.cc

```diff
@@ -1102,20 +1102,28 @@ google_firestore_v1_StructuredQuery_Filter Serializer::EncodeSingularFilter(
     const FieldFilter& filter) const {
   google_firestore_v1_StructuredQuery_Filter result{};
 
-  if (filter.op() == Filter::Operator::Equal) {
-    if (filter.value().is_null() || filter.value().is_nan()) {
-      result.which_filter_type =
-          google_firestore_v1_StructuredQuery_Filter_unary_filter_tag;
-      result.unary_filter.which_operand_type =
-          google_firestore_v1_StructuredQuery_UnaryFilter_field_tag;
-      result.unary_filter.field.field_path = EncodeFieldPath(filter.field());
+  if (filter.op() == Filter::Operator::Equal ||
+      filter.op() == Filter::Operator::NotEqual) {
+    result.which_filter_type =
+        google_firestore_v1_StructuredQuery_Filter_unary_filter_tag;
+    result.unary_filter.which_operand_type =
+        google_firestore_v1_StructuredQuery_UnaryFilter_field_tag;
+    result.unary_filter.field.field_path = EncodeFieldPath(filter.field());
+
+    if (filter.value().is_nan()) {
+      auto op =
+          filter.op() == Filter::Operator::Equal
+              ? google_firestore_v1_StructuredQuery_UnaryFilter_Operator_IS_NAN
+              : google_firestore_v1_StructuredQuery_UnaryFilter_Operator_IS_NOT_NAN;
```

Comment:
This line exceeds the maximum line length (<=80). Should I just move this into a variable to try and squeeze it in?